packages/edit-site/src/components/layout/style.scss

```diff
@@ -26,6 +26,13 @@
 	}
 }
 
+.edit-site-layout__header-container:has(+ .edit-site-layout__content > .edit-site-layout__mobile) {
```

Comment:
Depending on what component (sidebar or content) is displayed, the header sibling would be different:

- content frame: `.edit-site-layout__content > .edit-site-layout__mobile`
- sidebar frame: `.edit-site-layout__content > interface-navigable-region edit-site-layout__sidebar-region`